src/main/java/unicash/model/transaction/predicates/TransactionContainsAllKeywordsPredicate.java

```diff
@@ -0,0 +1,174 @@
+package unicash.model.transaction.predicates;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import unicash.commons.util.ToStringBuilder;
+import unicash.model.transaction.Transaction;
+
+
+/**
+ * Tests that the {@code Transactions}'s properties matches all the keywords given.
+ * Matching in this context either means a partial match of the keyword with the query, or
+ * a full match of the keyword, repeated for each keyword present in the list of keywords.
+ *
+ * </p> The matching depends on the property of the Transaction being matched, and the
+ * most appropriate matching is specified within the associated property predicate. This class
+ * simulates a composed predicate that represents a short-circuiting logical AND of all property
+ * predicates.
+ *
+ * </p> Encapsulated within is a list of transaction predicates, and this list can be accessed
+ * publicly, and modified. The test method returns true only if the input Transaction matches
+ * all predicates in this list.
+ */
+public class TransactionContainsAllKeywordsPredicate implements Predicate<Transaction> {
+
+    private List<Predicate<Transaction>> predicateList;
+
+    public TransactionContainsAllKeywordsPredicate() {
+        this.predicateList = new ArrayList<>();
+    }
+
+    public TransactionContainsAllKeywordsPredicate(List<Predicate<Transaction>> predicateList) {
+        this.predicateList = predicateList;
+    }
```

Comment:
Alternatively can consider using the `this()` call to initialize to another constructor

Reply:
Nice, I didn't catch that, thanks